sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py

```diff
@@ -852,6 +1088,46 @@ def _call_action(self, goals):
                 self._clients[client].send_goal(
                     goals[client], lambda terminal_state, result: self._action_done_cb(client, terminal_state, result))
 
+    def _set_point_cb(self, msg):
+        """
+        Updates the dictionary mapping joint names with their desired position in the trajectory controllers
+        @param msg - ROS message of type JointTrajectoryControllerState
+        """
+        joint_names_group = self._move_group_commander.get_active_joints()
+        with self._set_points_lock:
+            for index, joint in enumerate(msg.joint_names):
+                if joint not in joint_names_group:
+                    continue
+                self._set_points.update({joint: msg.desired.positions[index]})
+            if not self._are_set_points_ready:
+                with self._set_points_cv:
+                    for joint_name in joint_names_group:
+                        if self._is_joint_underactuated(joint_name) and f"{joint_name[0:5]}J0" not in joint_names_group:
+                            joint_names_group.append(f"{joint_name[0:5]}J0")
+                    for joint_name in joint_names_group:
+                        if joint_name not in self._set_points.keys():
+                            break
+                    else:
+                        self._are_set_points_ready = True
+                        self._set_points_cv.notifyAll()
+
+    def _set_point_j0_cb(self, msg, joint_name):
```

Comment:
I think we took the decision of listening to J0's set points because the J1 and J2 set points because of the underactuation might be far from the real state and also outside of the joint limits. For instance, when the set point of FFJ2 is 90deg and the set point of FFJ1 is ~60deg I find this:
![image](https://user-images.githubusercontent.com/62122095/207287150-d4fca50c-2014-4d9a-a212-350a51a02c79.png)
If I directly set these set points as the target state when planning, moveit will throw the error: target state is outside of the bounds. This also depends on a tolerance that I can't find right now, but I will and update this.
However, since we also encountered the "problem" of the set points being far from the state and added the "fix" of checking if the set point is far from the state and if it is, use the state then listening to j0's might not be useful.
I also agree that perhaps we should rethink of this whole architecture again.